controllers/azurejson_machinetemplate_controller.go

```diff
@@ -106,12 +104,8 @@ func (r *AzureJSONTemplateReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result
 		Name:      cluster.Spec.InfrastructureRef.Name,
 	}
 
-	err = r.Get(ctx, azureClusterName, azureCluster)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			log.Info("object was not found")
-			return reconcile.Result{}, nil
-		}
+	if err := r.Get(ctx, azureClusterName, azureCluster); err != nil {
```

Comment:
the same logic exists in the other azurejson_ controllers, do they also need to be fixed?

Reply:
yeah good catch. if it's okay since the logic is the same, i'm only going to write the test once (it'd be identical across all three).

another reason to figure out a way to clean up the preamble a bit...

Reply:
added

Reply:
meh, might as well add copies for the other reconcilers

Reply:
yeah let's combine the tests when we combine the code.  Until then nothing is guaranteed...